lib/filterx/object-datetime.c

```diff
@@ -205,6 +205,38 @@ _repr(FilterXObject *s, GString *repr)
   return datetime_repr(&ut, repr);
 }
 
+static FilterXObject *
+_add(FilterXObject *lhs_object, FilterXObject *rhs_object)
+{
+  if (!filterx_object_is_type(lhs_object, &FILTERX_TYPE_NAME(datetime)))
+    goto error;
+
+  UnixTime result;
+  UnixTime base = filterx_datetime_get_value(lhs_object);
+  if (filterx_object_is_type(rhs_object, &FILTERX_TYPE_NAME(datetime)))
+    {
+      UnixTime add = filterx_datetime_get_value(rhs_object);
+      result = unix_time_add_unix_time(base, add);
+    }
+  else if (filterx_object_is_type(rhs_object, &FILTERX_TYPE_NAME(integer)))
+    {
+      GenericNumber gn = filterx_primitive_get_value(rhs_object);
+      result = unix_time_add_unix_epoch(base, gn_as_int64(&gn));
+    }
+  else if (filterx_object_is_type(rhs_object, &FILTERX_TYPE_NAME(double)))
+    {
+      GenericNumber gn = filterx_primitive_get_value(rhs_object);
+      result = unix_time_add_unix_epoch(base, (gint64)(gn_as_double(&gn) * USEC_PER_SEC));
+    }
+  else
+    goto error;
+
+
+  return filterx_datetime_new(&result);
+error:
```

Comment:
[optional]

Using goto for a simple `return NULL` seems unnecessary, but I understand the general pattern so it is completely okay to keep it like this.